src/test/java/com/michelin/ns4kafka/controller/NamespaceControllerTest.java

```diff
@@ -371,4 +375,113 @@ void shouldNotDeleteNamespaceWhenResourcesAreStillLinkedWithIt() {
+    }
+
+    @Test
+    void shouldDeleteNamespacesInDryRunMode() {
+        Namespace existing1 = Namespace.builder()
+                .metadata(Metadata.builder()
+                        .name("namespace1")
+                        .cluster("local")
+                        .build())
+                .spec(Namespace.NamespaceSpec.builder()
+                        .kafkaUser("user")
+                        .build())
+                .build();
+        Namespace existing2 = Namespace.builder()
+                .metadata(Metadata.builder()
+                        .name("namespace2")
+                        .cluster("local")
+                        .build())
+                .spec(Namespace.NamespaceSpec.builder()
+                        .kafkaUser("user")
+                        .build())
+                .build();
+
+        when(namespaceService.findByWildcardName("namespace*"))
+                .thenReturn(List.of(existing1, existing2));
+        when(namespaceService.findAllResourcesByNamespace(existing1))
+                .thenReturn(List.of());
+        when(namespaceService.findAllResourcesByNamespace(existing2))
+                .thenReturn(List.of());
+
+        var result = namespaceController.bulkDelete("namespace*", true);
+        verify(namespaceService, never()).delete(any());
+        assertEquals(HttpResponse.noContent().getStatus(), result.getStatus());
+    }
+
+    @Test
+    void shouldNotDeleteNamespacesWhenResourcesAreStillLinkedWithIt() {
+        Namespace existing1 = Namespace.builder()
+                .metadata(Metadata.builder()
+                        .name("namespace1")
+                        .cluster("local")
+                        .build())
+                .spec(Namespace.NamespaceSpec.builder()
+                        .kafkaUser("user")
+                        .build())
+                .build();
+        Namespace existing2 = Namespace.builder()
+                .metadata(Metadata.builder()
+                        .name("namespace2")
+                        .cluster("local")
+                        .build())
+                .spec(Namespace.NamespaceSpec.builder()
+                        .kafkaUser("user")
+                        .build())
+                .build();
+
+        when(namespaceService.findByWildcardName("namespace*"))
+                .thenReturn(List.of(existing1, existing2));
+        when(namespaceService.findAllResourcesByNamespace(existing1))
+                .thenReturn(List.of("Topic/topic1"));
+        when(namespaceService.findAllResourcesByNamespace(existing2))
+                .thenReturn(List.of());
+
+        assertThrows(ResourceValidationException.class,
+                () -> namespaceController.bulkDelete("namespace*", false));
+        verify(namespaceService, never()).delete(any());
+    }
+
+    @Test
+    void shouldNotDeleteNamespacesWhenPatternMatchesNothing() {
+        var result = namespaceController.bulkDelete("namespace*", false);
```

Comment:
```java
namespaceService.findByWildcardName(name);
```

is returning empty by default?

I'm just having some doubts, might be better to mock it explicitely...

Reply:
Methode mocked